File: dastanlib/src/main/java/com/dastanapps/dastanlib/utils/DrawableUtils.java
package com.dastanapps.dastanlib.utils;

import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.StateListDrawable;

public class DrawableUtils {

    public static StateListDrawable createSelector(Drawable normalIconDrawable, Drawable selectedIconDrawable) {
        StateListDrawable res = new StateListDrawable();
        res.addState(new int[]{ android.R.attr.state_selected }, selectedIconDrawable);
        res.addState(new int[]{}, normalIconDrawable);
        return res;
    }

    public static GradientDrawable createGradientDrawable(int[] colors) {
        GradientDrawable shape = new GradientDrawable(GradientDrawable.Orientation.LEFT_RIGHT, colors);
        shape.setShape(GradientDrawable.RECTANGLE);
        return shape;
    }
}

File: common/src/main/java/com/dastanapps/dastanlib/utils/DrawableUtils.kt
package com.dastanapps.dastanlib.utils

import android.graphics.drawable.Drawable
import android.graphics.drawable.GradientDrawable
import android.graphics.drawable.StateListDrawable

object DrawableUtils {

    fun createSelector(normalIconDrawable: Drawable, selectedIconDrawable: Drawable): StateListDrawable {
        val res = StateListDrawable()
        res.addState(intArrayOf(android.R.attr.state_selected), selectedIconDrawable)
        res.addState(intArrayOf(), normalIconDrawable)
        return res
    }

    fun createGradientDrawable(colors: IntArray): GradientDrawable {
        val shape = GradientDrawable(GradientDrawable.Orientation.LEFT_RIGHT, colors)
        shape.shape = GradientDrawable.RECTANGLE
        return shape
    }
}
